octobot_trading/personal_data/orders/states/fill_order_state.py

```diff
@@ -82,7 +82,8 @@ async def terminate(self):
 
             # compute trading fees
             try:
-                self.order.fee = self.order.get_computed_fee()
+                if self.order.exchange_manager is not None:
```

Comment:
We could but here we specifically want to check if there is an exchange manager as it's the only required element. To me it's clearer to check it this way, don't you think so ?

Reply:
If the reason why exchange_manager is None is due to order cleanup I think we should use  order.is_cleaned(). If not we shouldn't

Reply:
Currently in the bot it's the only reason it can be None so it's the reason.
Why I think it's better to leave the check with `order.exchange_manager` is because it's focused on what we want to check here to compute fees. There is no semantic link between the fact that an order is cleared or not and computing its fees. Even if in fact the result is the same, it doesn't semantically make sense to check if the order is cleared before computing its fees. If we want to avoid this `order.exchange_manager is not None`, to me we should create a semantically correct method, something like `order.can_compute_fees` or something like that.

Reply:
Ok thanks, it understand better now :+1: